### DISCUSSION TITLE
Sunset of Team Discussions

### DISCUSSION THREAD

**Discussion by wborn** [openhab/openhab-distro#1512] [openHAB Maintainers Discussions]
Hi @openhab/maintainers!

If you look at the [top of the page](https://github.com/orgs/openhab/teams/maintainers) there is a popup that these Team Discussions are deprecated. So we have to decide if all this content should be transferred to GitHub Discussions and update the [Governance docs](https://www.openhab.org/docs/developer/governance.html) for this. I haven't had the time to thoroughly investigate this but maybe someone else does have time for it. :wink:

**Comment by florian-h05** [comment]
I have already moved our openHAB JavaScript library maintainers‘ team discussions to GitHub discussions, they are now in the openhab-js repo‘s discussions (https://github.com/openhab/openhab-js/discussions/categories/team-posts). In general this looks good to me, but **we have to keep in mind that the GitHub discussions are always public**, which might be unsuitable for e.g. maintainer elections.

If we want private discussions, I think we could create a private repo where only maintainers have access to (and probably contributors as well) and use this repos discussions.

Anyway, if we want to continue some of our team discussions, we have to move them. If we don’t move them, IIRC they will become readonly and only be accessible as raw text.

**Comment by digitaldan** [comment]
>If we want private discussions, I think we could create a private repo where only maintainers have access to (and probably contributors as well) and use this repos discussions.

Thats a very good idea, i like keeping these discussions in github and not spread across multiple tools like slack or the community forums, although the later is a possible alternative i guess.

**Comment by kaikreuzer** [comment]
Our governance docs also state:
_"To operate with transparency, discussions, minutes, deliberations, project plans, plans for new features, and other artifacts **are open, public, and easily accessible**."_

So I don't see any issue that they are public - I guess the Github discussions are therefore a good place.
Should we have a try with the "transfer" button that Github offers for the move?

**Comment by wborn** [comment]
I clicked the button and it then asks what repo the discussions should be transferred to. So we also need to decide if it is an existing repo. If a new one should be created. If we want to move all the discussions of each team to this same repo.

![image](https://user-images.githubusercontent.com/12213581/224478307-90a94e16-8692-4b2c-a437-b5735d63edde.png)

**Comment by wborn** [comment]
Apparently you can configure what repo is used for organization discussions in the organization settings.

For openHAB you can find it here:

https://github.com/organizations/openhab/settings/discussions

Example:

![Screenshot](https://user-images.githubusercontent.com/12213581/224478125-ba1a4e60-1481-4af0-8ae8-62c82a830272.png)

**Comment by J-N-K** [comment]
May understanding is that we can have "per repo" and "per organization" discussions. Probably "openhab-distro" would be a good choice for the "organization wide" discussion and all repos can have one of their own for repo specific discussions.

**Comment by kaikreuzer** [comment]
Makes sense. We also use the distro repo for the release notes, so that is more or less the one that aggregates the rest - so the org discussions could reside there as well.

**Comment by wborn** [comment]
I see someone enabled the discussions on https://github.com/openhab/openhab-distro/discussions 

We probably also need to give the Contributors (or only Maintainers?) team write access for them to be able to create new discussions. Without that you can only comment on existing discussions. I've already updated the branch protection rules so they are similar to openhab-addons but it wouldn't hurt if @kaikreuzer can also double check the permissions before team permissions are changed. :slightly_smiling_face: 

Users who cannot create new discussions will probably also start hijacking existing ones to start their own unrelated discussions. :wink:

**Comment by wborn** [comment]
It might be easier to allow any user to start new discussions by enabling "Allow users with read access to create discussions" in the organization config:

https://github.com/organizations/openhab/settings/member_privileges

![Screenshot](https://user-images.githubusercontent.com/12213581/226868349-e71989c5-f10b-4c7e-9dee-d38e9f08d77c.png)

**Comment by kaikreuzer** [comment]
> I see someone enabled the discussions

Yeah, that was actually me yesterday, because I wanted to start a discussion among @openhab/distro-maintainers. I was then trying to figure out, how we could best keep discussions of @openhab/maintainers and @openhab/distro-maintainers apart, which doesn't feel to be easy.

> It might be easier to allow any user to start new discussions by enabling "Allow users with read access to create discussions" in the organization config

That was actually enabled by default and my fist action was to disable it so that people do not start using this instead of our community forum.
I am not clear on what users have "read-access" - my assumption was that for public repos, every GitHub user has read access and therefore this feature must not be activated. I'll have to dig a bit deeper into the permissioning, but probably won't find time before the weekend. If you have a clear view on what needs to be done so that only our defined maintainer teams get permission to create new discussions, let me know.

**Comment by wborn** [comment]
> That was actually enabled by default and my fist action was to disable it so that people do not start using this instead of our community forum.

Yes I also don't think it is a good idea that GitHub Discussions will compete with the Community as it will scatter discussions all over the place.

>  If you have a clear view on what needs to be done so that only our defined maintainer teams get permission to create new discussions, let me know.

There is a matrix that outlines what you can do regarding Discussions depending on your role:

https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/repository-roles-for-an-organization

So you need at least the "Triage" role to create new discussions if you don't enable "Allow users with read access to create discussions" in the organization config. 

Another way is to turn categories into "Announcement" then you need at least the Maintainer role so that is probably only useful if we use a seperate repo for Discussions:
![discussions](https://user-images.githubusercontent.com/12213581/227171247-34180f6d-31c5-43fb-9c95-09a68c213bca.png)

Regardless of most configurations, any GitHub user will be able to comment on any existing discussion so they can still hijack them for their own unrelated discussions. The only way to prevent this from happening is probably to use Discussions in a repo to which only Contributors/Maintainers have access.

**Comment by kaikreuzer** [comment]
Ok, I think I found a suitable solution.
I have added @openhab/maintainers, @openhab/architecture-council and @openhab/contributors  to the repo with triage permissions - this means that all these groups can create new topics in the [architecture council category](https://github.com/openhab/openhab-distro/discussions/categories/architecture-council-discussions) and the [maintainers category](https://github.com/openhab/openhab-distro/discussions/categories/architecture-council-discussions) (while we trust that @openhab/contributors won't create new topics in the latter).
The [distro maintainers category](https://github.com/openhab/openhab-distro/discussions/categories/distro-maintainers-discussions) is created as an "announcement" type, so that only repo admins have the right to create new topics in it - this should be the way to set up discussions for the specific maintainer groups in all our other repos as well.

Let's have a try if that works out as expected.

**Comment by wborn** [comment]
I just transferred the Core and Docker team discussions too. 
Do you want to try more before transferring all these Maintainers discussions @kaikreuzer?

**Comment by kaikreuzer** [comment]
No, that looks all fine - feel free to transfer any further ones (or leave it to the according repo maintainer teams).

**Reply by wborn** [reply]
We made it to the other side! :slightly_smiling_face:

**Comment by wborn** [comment]
There are still some team discussions that haven't been migrated.
It's not clear to me if any migrations are still possible after May 8th.

https://github.blog/changelog/2023-02-08-sunset-notice-team-discussions/

> GitHub.com Timeline:
> 
> * Feb 8, 2023: A banner to migrate will be visible to maintainers at the top of their Team Discussions page, with the migration tooling included.
> * May 8, 2023: Team Discussions will be deprecated.
> * After May 8, 2023: Access to unmigrated Team Discussions will be available as raw text, but the UI won’t be available to add, modify, or delete Team Discussions.

**Reply by mueller-ma** [reply]
I'd keep the discussions of @openhab/android-maintainers private as they are about access to signing keys or access to Play Store.

**Comment by mherwege** [comment]
I noticed it became impossible to enter a feature request issue on the openhab-ios repository. It takes you with a link to opening a discussion topic. I assume permissions are set to not allow creating topics for someone who is not already a contributor. I raise it here, as I don't know where to best raise this. I actually think a feature request should be an issue (so it can be properly linked to PR's as well). But the most important thing is to be able to add a feature request, and not only in the forum. That's where it would get lost.

**Comment by wborn** [comment]
> So we have to decide if all this content should be transferred to GitHub Discussions and update the [Governance docs](https://www.openhab.org/docs/developer/governance.html) for this

I just created a PR for updating the Governance docs (https://github.com/openhab/openhab-docs/pull/2120) and pinned some "Team rules" discussions so they stand out more.
